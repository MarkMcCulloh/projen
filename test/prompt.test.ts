import { directorySnapshot, execProjenCLI, withProjectDir } from './util';
import prompts from 'prompts';

test('creating node project with prompts', () => {
  withProjectDir(projectdir => {

    prompts.inject(['description', 'package-name', false])
    execProjenCLI(projectdir, ['new', 'node', '--prompt', '--no-synth']);

    const projenrc = directorySnapshot(projectdir)['.projenrc.js'];
    expect(projenrc).toMatchSnapshot();
  });
});
