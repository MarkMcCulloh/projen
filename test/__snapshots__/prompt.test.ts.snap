// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating node project with prompts 1`] = `
"const { javascript } = require(\\"projen\\");
const project = new javascript.NodeProject({
  defaultReleaseBranch: \\"main\\",
  description: \\"h\\",
  name: \\"my-project\\",
  packageName: \\"h\\",
  release: false,

  // deps: [],     /* Runtime dependencies of this module. */
  // devDeps: [],  /* Build dependencies for this module. */
});
project.synth();"
`;
